src/tbb/arena.h

```diff
@@ -179,6 +179,81 @@ class atomic_flag {
     }
 };
 
+#if __TBB_PREVIEW_PARALLEL_PHASE
+class thread_leave_manager {
+    static const std::uint64_t FAST_LEAVE = 1;
+    static const std::uint64_t ONE_TIME_FAST_LEAVE = 1 << 1;
+    static const std::uint64_t DELAYED_LEAVE = 1 << 2;
+    static const std::uint64_t PARALLEL_PHASE = 1 << 3;
+
+    std::atomic<std::uint64_t> my_state{0};
+public:
+    void set_initial_state(tbb::task_arena::leave_policy lp) {
```

Comment:
Since this is essentially the constructor, please add a comment that this method is required to be called soon after construction, and is not thread-safe. Or maybe consider converting it into a real constructor.